vraptor-core/src/main/java/br/com/caelum/vraptor/converter/PrimitiveByteArrayConverter.java

```diff
@@ -0,0 +1,73 @@
+/***
+ * Copyright (c) 2009 Caelum - www.caelum.com.br/opensource
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.caelum.vraptor.converter;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import java.nio.charset.Charset;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+import javax.interceptor.Interceptor;
+
+import br.com.caelum.vraptor.Convert;
+import br.com.caelum.vraptor.http.EncodingHandler;
+
+/**
+ * A Byte array converter. Null or empty values are returned as an empty byte array.
+ *
+ * @author Otávio Scherer Garcia
+ * @since 4.2.0
+ */
+@Convert(byte[].class)
```

Comment:
I'd vote to wait a little more users asking for it (maybe here, after seeing this pr)
that's because I've never seen someone asking for it on vraptor's mail list or something
you can easily package this converter in a jar that can be shared on your apps